lib/dart_2_party_ecdsa.dart

```diff
@@ -248,53 +248,13 @@ final class Dart2PartySDK {
 
   late final BackupState backupState = BackupState(localDatabase);
 
-  CancelableOperation<String> fetchRemoteBackup(String accountAddress, String userId) {
-    if (_state != SdkState.readyToSign) {
-      return CancelableOperation.fromFuture(Future.error(StateError('Cannot start backup when SDK in $_state state')));
-    }
-
-    final keyshare = keygenState.keysharesMap[METAMASK_WALLET_ID]?.firstWhereOrNull((keyshare) => keyshare.ethAddress == accountAddress);
-    if (keyshare == null) {
-      return CancelableOperation.fromFuture(Future.error(StateError('Cannot find keyshare for $accountAddress')));
-    }
-    final fetchBackupAction = FetchRemoteBackupAction(_sharedDatabase, userId);
-    final fetchBackupOperation = CancelableOperation.fromFuture(fetchBackupAction.start(), onCancel: fetchBackupAction.cancel);
-
-    return fetchBackupOperation.then((remoteBackup) {
-      if (remoteBackup.isNotEmpty) {
-        final accountBackup = AccountBackup(accountAddress, keyshare.toBytes(), remoteBackup);
-        backupState.upsertBackupAccount(METAMASK_WALLET_ID, accountBackup);
-      }
-      return remoteBackup;
-    });
-  }
-
   Stream<BackupMessage> listenRemoteBackup(String userId) {
     if (_state != SdkState.readyToSign) {
       throw StateError('Cannot start backup when SDK in $_state state');
     }
 
-    final remoteBackupListener = RemoteBackupListener(_sharedDatabase, userId);
-    return remoteBackupListener.remoteBackupRequests().tap((remoteBackup) {
-      if (remoteBackup.address != null && remoteBackup.walletId != null && remoteBackup.backupData.isNotEmpty) {
-        final accountAddress = remoteBackup.address ?? "";
-        final walletId = remoteBackup.walletId ?? "";
-
-        if (accountAddress.isNotEmpty && walletId.isNotEmpty) {
-          if (keygenState.keysharesMap[walletId] == null) {
-            throw StateError('No keyshares for $walletId');
-          }
-          final keyshare = keygenState.keysharesMap[walletId]!.firstWhereOrNull((keyshare) => keyshare.ethAddress == accountAddress);
-          if (keyshare == null) {
-            throw StateError('Cannot find keyshare for $accountAddress of $walletId provider');
-          }
-          final accountBackup = AccountBackup(accountAddress, keyshare.toBytes(), remoteBackup.backupData);
-          backupState.upsertBackupAccount(walletId, accountBackup);
-        }
-      }
-    }).handleError((error) {
-      print('Error listening remote backup: $error');
-    });
+    final remoteBackupListener = RemoteBackupListener(_sharedDatabase, userId, keygenState, backupState);
```

Comment:
Delete the backup document from firestore after phone reads the document

Reply:
https://github.com/silence-laboratories/silent-shard-flutter-sdk/blob/72636f03b239182faf8b661204a1cff14af1e73f/lib/dart_2_party_ecdsa.dart

Regarding the history of commits, I don't think we delete backup document before. And, our `Future<void> deleteBackupMessage(String userId) => _transport.delete("backup", userId);` will remove all documents with the same userId, it will effect Stackup, Biconomy backups too

Reply:
Ok, we can discuss later, not important for this release.